src/models/user_account.ts

```diff
@@ -36,10 +36,11 @@ export default class UserAccount extends Model<
   @Column
   public readonly updatedAt!: Date
 
-  public asProfile(isAdmin = false): UserProfile {
+  public asProfile(isAdmin = false, groupId?: number): UserProfile {
```

Comment:
Suggestion (blocking): swap these so the param with a default is second so the caller can omit it if they're ok with the default.